assertj-core/src/test/java/org/assertj/core/api/string_/StringAssert_asBoolean_Test.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.api.string_;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+import org.assertj.core.api.StringAssert;
+import org.assertj.core.api.StringAssertBaseTest;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for <code>{@link StringAssert#asBoolean()}</code>.
+ *
+ * @author HeZean
+ */
+class StringAssert_asBoolean_Test extends StringAssertBaseTest {
+
+  @Override
+  protected StringAssert invoke_api_method() {
+    assertions.asBoolean();
+    return null;
+  }
+
+  @Override
+  protected void verify_internal_effects() {
+    // Verify disabled as the asBoolean cast have no internal effect.
+  }
+
+  @Override
+  public void should_return_this() {
+    // Test disabled as the assertion does not return this.
+  }
+
+  @Test
+  void should_parse_string_as_true_when_given_a_true_like_string() {
+    // GIVEN
+    String string = "true";
+    // THEN
+    assertThat(string).asBoolean().isTrue();
+
+    // GIVEN
+    string = "\n\ttruE  ";
+    // THEN
+    assertThat(string).asBoolean().isTrue();
```

Comment:
replace `assertThat` with `then` from `BDDAssertions` for readability (and to follow the https://github.com/assertj/assertj/blob/main/CONTRIBUTING.md guidelines).
This applies to all the tests in this PR obviously :wink: